swagger.yaml

```diff
@@ -1,42 +1,254 @@
+
 definitions:
-  HelloResponse:
+
+  Job:
+    type: object
+    required:
+      - "id"
+      - "last_updated"
+      - "links"
+      - "search_index_name"
+      - "state"
+    properties:
+      id:
+        type: string
+      last_updated:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      links:
+        type: object
+        required:
+          - "tasks"
+          - "self"
+        properties:
+          tasks:
+            type: string
+            format: http://localhost:<port>/jobs/{id}/tasks
+          self:
+            type: string
+            format: http://localhost:<port>/jobs/{id}
+      number_of_tasks:
+        type: integer
+      reindex_completed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_failed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_started:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      search_index_name:
+        type: string
+      state:
+        type: string
+        enum:
+          - CREATED
+          - IN-PROGRESS
+          - COMPLETED
+          - FAILED
+      total_search_documents:
+        type: integer
+
+  NewJob:
     type: object
+    required:
+      - "last_updated"
+      - "search_index_name"
+      - "state"
     properties:
-      message:
+      last_updated:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      number_of_tasks:
+        type: integer
+      reindex_completed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_failed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_started:
         type: string
-        description: "Message returned by hello world endpoint"
-        example: "Hello, world!"
\ No newline at end of file
+        format: date
+        description: ISO8601 timestamp
+      search_index_name:
+        type: string
+      state:
+        type: string
+        enum:
+          - CREATED
+          - IN-PROGRESS
+          - COMPLETED
+          - FAILED
+      total_search_documents:
+        type: integer
+
+  Jobs:
+    type: object
+    required:
+      - "count"
+      - "items"
+      - "limit"
+      - "offset"
+      - "total_count"
+    properties:
+      count:
+        type: integer
+      items:
+        type: array
+        items:
+          $ref: '#/definitions/Job'
+      limit:
+        type: integer
+      offset:
+        type: integer
+      total_count:
+        type: integer
+      sort:
+        type: integer
+
+parameters:
+
+  id:
+    name: id
+    description: The id of a search reindex job when job POSTed (Job.id)
+    in: path
+    required: true
+    type: string
+
+  limit:
+    name: limit
+    description: The maximum number of Job objects we're returning in this Jobs object. The default limit is 20.
```

Comment:
I think it is better to refer to a job as a `resource` instead of an `object`

Reply:
done